presto-main/src/main/java/io/prestosql/security/AccessControl.java

```diff
@@ -46,6 +46,35 @@
      */
     void checkCanImpersonateUser(Identity identity, String userName);
 
+    /**
+     * Checks if identity can execute a query.
+     *
+     * @throws AccessDeniedException if not allowed
+     */
+    void checkCanExecuteQuery(Identity identity);
+
+    /**
+     * Checks if identity can view a query owned by the specified user.  The method
+     * will not be called when the current user is the query owner.
+     *
+     * @throws AccessDeniedException if not allowed
+     */
+    void checkCanViewQueryOwnedBy(Identity identity, String queryOwner);
```

Comment:
I don't think we should make that change now.  I don't think we have enough information to design this for the future (e.g., how is it going to work when we implement proper SQL roles).  For the specific use case, the implementer of such a system could simply pull the groups again, which they may want to do anyway, since the group membership would be old.